airbyte-integrations/connectors/source-stripe/source_stripe/partition.py

```diff
@@ -0,0 +1,130 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import copy
+from typing import Any, Callable, Iterable, Mapping, Optional, Union
+
+import requests
+from airbyte_cdk.sources.declarative.extractors import RecordSelector
+from airbyte_cdk.sources.declarative.requesters import Requester
+from airbyte_cdk.sources.declarative.requesters.paginators import DefaultPaginator
+from airbyte_cdk.sources.declarative.requesters.request_option import RequestOptionType
+from airbyte_cdk.sources.message import MessageRepository
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.concurrent.partitions.partition import Partition
+from airbyte_cdk.sources.streams.concurrent.partitions.partition_generator import PartitionGenerator
+from airbyte_cdk.sources.streams.concurrent.partitions.record import Record
+from airbyte_protocol.models import SyncMode
+
+
+class PaginatedRequester:
+    def __init__(self, requester: Requester, record_selector: RecordSelector, paginator: DefaultPaginator):
+        self._requester = requester
+        self._record_selector = record_selector
+        self._page_token_option = paginator.page_token_option
```

Comment:
why create a new field instead of always referring to `paginator.page_token_option`?